donkeycar/tests/mavlink_test.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python3
+from __future__ import print_function
+"""
+Script for using a Pixhawk running any ArduPilot- or PX4-based code with Donkeycar 
+
+author: @zlite (Chris Anderson), 2022
+contrib: @wallarug (Cian Byrne) 2019
+contrib: @peterpanstechland 2019
+contrib: @sctse999 2020
+
+If you're getting a permissions error on the USB port: "sudo chmod 666 /dev/ttyACM0"
+
+MAVlink messages:
+read RC PWM: RC_CHANNELS
+read attitude: ATTITUDE
+read GPS: GLOBAL_POSITION_INT
+read battery: BATTERY_STATUS
+read encoder: WHEEL_DISTANCE
+
+write RC PWM: MAV_CMD_DO_SET_SERVO
+
+"""
+
+import time
+import donkeycar as dk
+import config as cfg
+
+debug = True
+
+MAX_FORWARD = cfg.MAVLINK_MAX_FORWARD
+MAX_REVERSE = cfg.MAVLINK_MAX_REVERSE
+STOPPED_PWM = cfg.MAVLINK_STOPPED_PWM
+STEERING_MID = cfg.MAVLINK_STEERING_MID
+angle = 0.0
+throttle = 0.0
+SHOW_STEERING_VALUE = cfg.MAVLINK_SHOW_STEERING_VALUE
+DEAD_ZONE = cfg.JOYSTICK_DEADZONE
+
+
+"""
+    Steering and throttle should range between -1.0 to 1.0. This function will
+    trim value great than 1.0 or less than 1.0
+"""
+
+
+try:
+    import serial
+except ImportError:
+    print("PySerial not found.  Please install: pip install pyserial")
+
+try:
+    from pymavlink import mavutil
```

Comment:
This test probably doesn't make sense as a pytest, if it can only be run where the corresponding part can be installed as well. Better to make this a `main()` method in the `mavlink.py` module.